tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -2091,15 +2105,13 @@ Status ConvertConv2DHelper(const OpConverterParams* params, int group,
     conv_layer = layer;
   }
 
-  // After creating the conv layer, if we are in explicit precision mode and the
-  // weights input is a tensor, then we need to override the weights input by
-  // calling setInput() on the layer.
-  if (params->use_explicit_precision) {
-    TRT_ENSURE(inputs.at(1).is_tensor());
-
-    nvinfer1::IShuffleLayer* layer = params->converter->network()->addShuffle(
-        *inputs.at(1).tensor()->trt_tensor());
-    layer->setFirstTranspose({3, 2, 0, 1});
+  // If we are in explicit Q/DQ mode (indicated by checking if the second input
+  // is a tensor), we need to override the weights input by calling setInput()
+  // on the layer.
+  if (inputs.at(1).is_tensor()) {
+    TRT_ENSURE(params->use_explicit_precision);
+    nvinfer1::IShuffleLayer* layer =
+        params->converter->network()->addShuffle(*filter_tensor->trt_tensor());
```

Comment:
This change does 2 things: 
1. Change checking explicit precision mode to read if the second input is a tensor rather than using the params (see PR description for more details)
2. Remove the transpose of the second input from here to further above so that we do not fail building the cuda engine.